includes/admin/marketplace-suggestions/class-wc-marketplace-updater.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Marketplace suggestions updater
+ *
+ * Uses WC_Queue to ensure marketplace suggestions data is up to date and cached locally.
+ *
+ * @package WooCommerce\Classes
+ * @since   3.6.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Marketplace Suggestions Updater
+ */
+class WC_Marketplace_Updater {
+
+	/**
+	 * Setup.
+	 */
+	public static function load() {
+		add_action( 'init', array( __CLASS__, 'init' ) );
+	}
+
+	/**
+	 * Schedule events and hook appropriate actions.
+	 */
+	public static function init() {
+		$queue = WC()->queue();
+		$next  = $queue->get_next( 'woocommerce_update_marketplace_suggestions' );
+		if ( ! $next ) {
+			$queue->schedule_recurring( time(), WEEK_IN_SECONDS, 'woocommerce_update_marketplace_suggestions' );
+		}
+
+		add_action( 'woocommerce_update_marketplace_suggestions', array( __CLASS__, 'update_marketplace_suggestions' ) );
+	}
+
+	/**
+	 * Fetches new marketplace data, updates wc_marketplace_suggestions.
+	 */
+	public static function update_marketplace_suggestions() {
+		$data = get_option(
```

Comment:
Why use `get_option` rather than transient?

Reply:
There is no benefit to using a transient since we already control the cache using the action scheduler. Plus with an option we guarantee the data availability because it's going to be stored in the database, regardless of any persistent object caching, LRU evictions, transient cleaners, etc.